src/t8_data/t8_vector_handler.hxx

```diff
@@ -0,0 +1,276 @@
+   */
+  t8_vector_handler (const std::vector<TType> &data)
+    : m_data (std::make_shared<std::vector<TType>> (data)), single_handler ()
+  {
+  }
+
+  /**
+   * Get the data.
+   * 
+   * \return std::shared_ptr<std::vector<TType>> 
+   */
+  std::shared_ptr<std::vector<TType>>
+  get_data () const
+  {
+    return m_data;
+  }
+
+  int
+  buffer_size (sc_MPI_Comm comm) override
+  {
+    int total_size = 0;
+    const int mpiret = sc_MPI_Pack_size (1, sc_MPI_INT, comm, &total_size);
+    SC_CHECK_MPI (mpiret);
+    if (m_data) {
+      total_size += std::accumulate (m_data->begin (), m_data->end (), 0, [&] (int sum, const TType &item) {
+        return sum + single_handler.size (item, comm);
+      });
+    }
+    return total_size;
+  }
+
+  void
+  pack_vector_prefix (void *buffer, const int num_bytes, int &pos, sc_MPI_Comm comm) override
+  {
+    const int num_data = m_data->size ();
+    const int mpiret = sc_MPI_Pack (&num_data, 1, sc_MPI_INT, buffer, num_bytes, &pos, comm);
+    SC_CHECK_MPI (mpiret);
+
+    std::for_each (m_data->begin (), m_data->end (),
+                   [&] (const TType &item) { single_handler.pack (item, pos, buffer, num_bytes, comm); });
+  }
+
+  void
+  unpack_vector_prefix (const void *buffer, const int num_bytes, int &pos, int &outcount, sc_MPI_Comm comm) override
+  {
+    const int mpiret = sc_MPI_Unpack (buffer, num_bytes, &pos, &outcount, 1, sc_MPI_INT, comm);
+    SC_CHECK_MPI (mpiret);
+    T8_ASSERT (outcount >= 0);
+
+    if (!m_data) {
+      m_data = std::make_shared<std::vector<TType>> (outcount);
+    }
+    else {
+      m_data->clear ();
+      m_data->resize (outcount);
+    }
+    std::for_each (m_data->begin (), m_data->end (),
+                   [&] (TType &item) { single_handler.unpack (buffer, num_bytes, pos, item, comm); });
+  }
+
+  int
+  send (const int dest, const int tag, sc_MPI_Comm comm) override
+  {
+    int pos = 0;
+    const int num_bytes = buffer_size (comm);
+    std::vector<char> buffer (num_bytes);
+    pack_vector_prefix (buffer.data (), num_bytes, pos, comm);
+
+    const int mpiret = sc_MPI_Send (buffer.data (), num_bytes, sc_MPI_PACKED, dest, tag, comm);
+    SC_CHECK_MPI (mpiret);
+    return mpiret;
+  }
+
+  int
+  recv (const int source, const int tag, sc_MPI_Comm comm, sc_MPI_Status *status, int &outcount) override
+  {
+    int pos = 0;
+    int mpiret = sc_MPI_Probe (source, tag, comm, status);
+    SC_CHECK_MPI (mpiret);
+
+    int num_bytes;
+    mpiret = sc_MPI_Get_count (status, sc_MPI_PACKED, &num_bytes);
+    SC_CHECK_MPI (mpiret);
+    std::vector<char> buffer (num_bytes);
+
+    mpiret = sc_MPI_Recv (buffer.data (), num_bytes, sc_MPI_PACKED, source, tag, comm, status);
+    SC_CHECK_MPI (mpiret);
+    unpack_vector_prefix (buffer.data (), num_bytes, pos, outcount, comm);
+    return mpiret;
+  }
+
+  constexpr t8_data_handler_type
+  type () override
+  {
+    return t8_data_handler_type (single_handler.type ());
+  }
```

Comment:
I know that there are docstrings in the abstract base, but these will only be shown if the object is casted into a pointer of that object. This class will not show any docstrings